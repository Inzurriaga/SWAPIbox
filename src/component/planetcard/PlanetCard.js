import React, { Component } from "react";

export default class PlanetCard extends Component{
    render(){
        return(
            <div className="planet-card">
                <h3>{this.props.planet.name}</h3>
                <p>Terrain: {this.props.planet.terrain}</p>
                <p>Population: {this.props.planet.population}</p>
                <p>Climate: {this.props.planet.climate}</p>Residents: 
                {
                    this.props.planet.residents.map((resident) => {
                        return (<p>{resident}</p>)
                    })
                }
            </div>
        )
    }
}