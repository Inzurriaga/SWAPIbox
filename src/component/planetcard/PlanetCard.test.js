import React from "react"
import { shallow } from "enzyme"
import PlanetCard from "./PlanetCard"

const mockdata = {
    name: "earth",
    terrain: "rocking",
    population: "3000",
    residents: []
}


describe("planet_card", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(
            <PlanetCard planet={mockdata}/>
        )
    })
    it("should match snapshot", () => {
        expect(wrapper).toMatchSnapshot()
    })
})