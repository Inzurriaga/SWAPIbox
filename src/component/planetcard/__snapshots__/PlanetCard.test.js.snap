// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`planet_card should match snapshot 1`] = `
<div
  className="planet-card"
>
  <h3>
    earth
  </h3>
  <p>
    Terrain: 
    rocking
  </p>
  <p>
    Population: 
    3000
  </p>
  <p>
    Climate: 
  </p>
  Residents:
</div>
`;
